src/components/enterprise-user-subsidy/data/hooks.jsx

```diff
@@ -11,39 +11,120 @@ import { LICENSE_STATUS } from './constants';
+ */
+const requestAutoAppliedUserLicense = async (customerAgreementId) => {
+  try {
+    const response = await requestAutoAppliedLicense(customerAgreementId);
+    const license = camelCaseObject(response.data);
+    return license;
+  } catch (error) {
+    logError(error);
+    return null;
+  }
+};
+
+/**
+ * Retrieves a license for the authenticated user, if applicable. First attempts to find any existing licenses
+ * for the user. If a license is found, the app uses it; otherwise, if the enterprise has an SSO/LMS identity
+ * provider configured and the customer agreement has a subscription plan suitable for auto-applied licenses,
+ * attempt to auto-apply a license for the user.
+ *
+ * @param {object} args
+ * @param {object} args.enterpriseConfig The enterprise customer config
+ * @param {object} args.customerAgreementConfig The customer agreement config associated with the enterprise
+ * @param {boolean} args.isLoadingCustomerAgreementConfig Whether the customer agreement is still resolving
+ * @returns Array containing a user license, if applicable, and whether the license data is still resolving
+ */
+export function useSubscriptionLicense({
+  enterpriseConfig,
+  customerAgreementConfig,
+  isLoadingCustomerAgreementConfig,
+}) {
   const [license, setLicense] = useState();
   const [isLoading, setIsLoading] = useState(true);
 
+  const {
+    uuid: enterpriseId,
+    identityProvider: enterpriseIdentityProvider,
+  } = enterpriseConfig;
+
   useEffect(() => {
-    fetchSubscriptionLicensesForUser(enterpriseId)
-      .then((response) => {
-        const { results } = camelCaseObject(response.data);
-        const activated = results.filter(result => result.status === LICENSE_STATUS.ACTIVATED);
-        const assigned = results.filter(result => result.status === LICENSE_STATUS.ASSIGNED);
-        const revoked = results.filter(result => result.status === LICENSE_STATUS.REVOKED);
-
-        if (activated.length) {
-          setLicense(activated.shift());
-        } else if (assigned.length) {
-          setLicense(assigned.shift());
-        } else if (revoked.length) {
-          setLicense(revoked.shift());
-        } else {
-          setLicense(null);
+    async function retrieveUserLicense() {
+      let result = await fetchExistingUserLicense(enterpriseId);
+
+      if (features.ENABLE_AUTO_APPLIED_LICENSES) {
+        const customerAgreementMetadata = [
+          customerAgreementConfig?.uuid,
+          customerAgreementConfig?.subscriptionForAutoAppliedLicenses,
+        ];
+        const hasCustomerAgreementData = customerAgreementMetadata.every(item => !!item);
+
+        // Per the product requirements, we only want to attempt requesting an auto-applied license
+        // when the enterprise customer has an SSO/LMS provider configured.
+        if (!result && enterpriseIdentityProvider && hasCustomerAgreementData) {
+          result = await requestAutoAppliedUserLicense(customerAgreementConfig?.uuid);
```

Comment:
nit: not sure if it still yells but should be able to use customerAgreementConfig.uuid here?

Reply:
Technically, yes. Leftover from copy/pasta. Will remove.